schema.ts

```diff
@@ -1256,3 +1256,15 @@ export function makeReferenceSchemaFromObjectSchema(
     properties: referenceProperties,
   });
 }
+
+/**
+ * Convenience for defining the result schema for an action. The identity enables Coda to
+ * update the corresponding sync table row, if it exists.
+ * You could add the identity directly, but that would make the schema less re-usable.
+ */
+export function withIdentity(schema: GenericObjectSchema, identityName: string): GenericObjectSchema {
+  return makeObjectSchema({
+    ...schema,
```

Comment:
Do we want to deep copy the schema, in case the maker mutates it?

Reply:
Oh, duh. Thanks! Done.